src/sync/stages/index/indexers/core/utxo_by_txo_ref.rs

```diff
@@ -177,4 +184,55 @@ impl UtxoCache {
             self.cache.weighted_size(),
         )
     }
+
+    pub fn wipe_if_mem_usage_high(&mut self) {
+        // Only check memory usage every 300 seconds to avoid overhead
+        let now = Instant::now();
+        if now.duration_since(self.last_check).as_secs() < 300 {
+            return;
+        }
+        self.last_check = now;
+
+        let mut system = System::new_all();
+
+        system.refresh_memory();
+
+        let total_memory = system
```

Comment:
This can be very tricky with shared instances in a kubernetes cluster

Reply:
Yeah thats what cgroup_limits does AFAICT